src/components/cross-section-component.tsx

```diff
@@ -65,6 +67,14 @@ export class CrossSectionComponent extends BaseComponent<IProps, IState>{
 
   public componentDidUpdate(prevProps: IProps) {
     this.recomputeMetrics();
+
+    if (this.props.isSelectingCrossSection !== prevProps.isSelectingCrossSection) {
+      this.stores.setCrossSectionSelectorVisibility(this.props.isSelectingCrossSection);
```

Comment:
Dispatching an action in `componentDidUpdate` is valid, but if there is another way to do it, I would.

`props.isSelectingCrossSection` is passed as a prop from `app.tx` which is reading that value from the store.  

Is there a way to handle these cases in the stores actions or possibly a computed property?